src/pdl/pdl_dumper.py

```diff
@@ -79,6 +80,18 @@ def block_to_dict(block: pdl_ast.BlockType, json_compatible: bool) -> DumpedBloc
         return block
     d: dict[str, Any] = {}
     d["kind"] = str(block.kind)
+    if block.start_nanos != 0:
+        d["start_nanos"] = block.start_nanos
+        d["end_nanos"] = block.end_nanos
+
+        now = datetime.datetime.now()
+        local_now = now.astimezone()
+        local_tz = local_now.tzinfo
+        if local_tz is not None:
+            local_tzname = local_tz.tzname(local_now)
+        else:
+            local_tzname = "UTC"
+        d["timezone"] = local_tzname
```

Comment:
What about computing the time zone once and setting it up in the interpreter in the `step_advanced_block_timed` function?

Reply:
i'm not sure if there is a great solution to this. either we do it outside of `block_to_dict()`, but then we would either need to pass this value in as a parameter, or we would need to store the timezone as a global variable. by design, it seems that the trace format does not allow for a global wrapper, where we could compute and store the timezone (and any other trace metadata, e.g. who created it, when it was created, etc.).

an alternate solution would be to store the start and end timestamps as formatted timestamps (this is how kubernetes does it), so that the timezone is encoded into each timestamp.